packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.tsx

```diff
@@ -178,14 +183,19 @@ export class ApplicationExtension<
                 )
             }
 
-            const component = componentMap[componentName]
-
-            if (!component) {
+            const Component = componentMap[componentName]
+            if (!Component) {
                 throw new Error(
                     `"${componentName}" was not found in the component map. Ensure that getComponentMap() includes a mapping for it in the ${this.getName()} extension`
                 )
             }
 
+            let component: React.ComponentType<any> = (props: Record<string, string>) => (
+                <Component {...props} {...route.componentProps} />
+            )
+            component = hoistNonReactStatics(component, Component)
+            component.displayName = componentName
```

Comment:
when we deserialize we need to ensure
1. the serialized props are injected into the component
2. the non react statics are copied over to the wrapped component
3. the displayName is set correctly